flink-core/src/test/java/org/apache/flink/core/memory/MemorySegmentTestBase.java

```diff
@@ -1955,11 +1955,6 @@ private void testByteBufferGet(boolean directBuffer) {
 		assertArrayEquals(bytes, result);
 	}
 
-	@Test(expected = ReadOnlyBufferException.class)
-	public void testHeapByteBufferGetReadOnly() {
```

Comment:
why is this test duplicated and moved to subclasses?

Reply:
Please note that the expected exception types are different for different sub-classes. For HybridOnHeapMemorySegmentTest, the type is IllegalArgumentException, and for HybridOffHeapMemorySegmentTest, the type is ReadOnlyBufferException

Reply:
Does the exception type not depend on the type of target buffer (input arg)?
It looks like it is the same `IllegalArgumentException` for both test subclasses, also in the PR.
Would it not be sufficient just to change `ReadOnlyBufferException` to `IllegalArgumentException`?

Reply:
Sorry, there is a mistake in my previous comment. 
The test in HeapMemorySegmentTest throws ReadOnlyBufferException. 
I have revised the code accordingly. Please take a look.